From d4abcf3999ee5141805ab70bf925983b336d49f3 Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Fri, 11 May 2012 16:49:56 +0100
Subject: [PATCH] Hide GList-based contact info typedefs from g-ir-scanner

Recent vapigen versions don't like the result.

This means we can revert the version that adds stub doc-comments for
them, which seems to confuse gtk-doc. Good thing we don't have several
duelling C-and-comment parsers or anything...

Bug: https://bugs.freedesktop.org/show_bug.cgi?id=49637
Signed-off-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
Reviewed-by: Xavier Claessens <xavier.claessens@collabora.co.uk>
---
 telepathy-glib/connection.h              |    8 ++++++++
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/telepathy-glib/connection.h b/telepathy-glib/connection.h
index 04506c0..b5af72b 100644
--- a/telepathy-glib/connection.h
+++ b/telepathy-glib/connection.h
@@ -48,7 +48,11 @@ TpContactInfoFieldSpec *tp_contact_info_field_spec_copy (
     const TpContactInfoFieldSpec *self);
 void tp_contact_info_field_spec_free (TpContactInfoFieldSpec *self);
 
+#ifndef __GI_SCANNER__
+/* the typedef only exists for G_DEFINE_BOXED_TYPE's benefit, and
+ * g-ir-scanner 1.32.1 doesn't parse a skip annotation */
 typedef GList TpContactInfoSpecList;
+#endif
 
 #define TP_TYPE_CONTACT_INFO_SPEC_LIST (tp_contact_info_spec_list_get_type ())
 GType tp_contact_info_spec_list_get_type (void);
@@ -73,7 +77,11 @@ TpContactInfoField *tp_contact_info_field_new (const gchar *field_name,
 TpContactInfoField *tp_contact_info_field_copy (const TpContactInfoField *self);
 void tp_contact_info_field_free (TpContactInfoField *self);
 
+#ifndef __GI_SCANNER__
+/* the typedef only exists for G_DEFINE_BOXED_TYPE's benefit, and
+ * g-ir-scanner 1.32.1 doesn't parse a skip annotation */
 typedef GList TpContactInfoList;
+#endif
 
 #define TP_TYPE_CONTACT_INFO_LIST (tp_contact_info_list_get_type ())
 GType tp_contact_info_list_get_type (void);
-- 
1.7.10

